src/Components/Blueprints/BlueprintActionsMenu.tsx

```diff
@@ -28,7 +33,10 @@ export const BlueprintActionsMenu: React.FunctionComponent<
   };
   const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
   const navigate = useNavigate();
-
+  const { data: blueprintDetails } = useGetBlueprintQuery({
+    id: selectedBlueprintId || '',
+  });
```

Comment:
This will fetch the details even when the user doesn't need it,
How about using `useLazyQuery`? 
```js
  const [trigger, { data, error, isLoading }] = useLazyQuery(api.endpoints.getBlueprint);
```

The `trigger` can be invoked in the onClickHandler, after the user clicks for downloading the blueprint file